README.md

```diff
@@ -1,2 +1,115 @@
+AUTH_SIGN_IN_URL | Authentication sign in URL | `https://cola.service.cabinetoffice.gov.uk/v2/<YOUR_SERVICE>/login`
+COOKIE_ID_NAME | The name of the cookie | `github-requests` for [github-requests-app](https://github.com/cabinetoffice/github-requests-app)
+COOKIE_PARSER_SECRET | Secret used in validating/calculating the cookie signature | `secret`
+COOKIE_SESSION_SECRET | Secret key for signing the session cookie | `secret`
+LOG_LEVEL       | Logging levels        | `info` (Log only if info.level is less than or equal to this level)
+HUMAN | Formatting messages form (default JSON) | `true` (Enable human formatting for log messages)
+
+## Files Structure
+
+Directory Path | Description
+--- | ---
+`./config/` | This folder contains configuration settings for the login module. It is essential for fetching sensitive information such as `COOKIE_PARSER_SECRET` or `COOKIE_SESSION_SECRET`.
+`./util/` | This folder contains utility functions used throughout the login module, including Session and Cookie utils.
+`./index.ts` | This file will expose the authentication middleware and Session and Cookie utils methods.
+`./test` | Jest Test files (`*.spec.ts`, `setup.ts`, and `*.mocks.ts`)
+Others files | Other files related to modules dependency, CI/CD, *git, lint, test/typescript configs …
+
+## Integration and Getting Started
+
+To integrate our login solution into your Node.js application, follow these simple steps:
+
+- **Install Dependencies**: Begin by installing the necessary dependencies using npm. Ensure that you have `express`, `cookie-parser` and `cookie-session` installed in your project.
+
+```sh
+npm i @co-digital/login
+```
+
+- **Middleware Configuration**: Configure the module to utilize the associated authentication middleware by incorporating the appropriate environment data and enabling `cookieParser` and `cookieSession` within the application configuration file.
+
+```ts
+// On app.ts
+...
+app.use(cookieParser(COOKIE_PARSER_SECRET));
+app.use(cookieSession({ secret: COOKIE_SESSION_SECRET }));
+...
+```
+
+- **Route Setup**: Define routes for any authentication-related endpoints required by your application and utilize the middleware to protect routes that require authentication.
+
+```ts
+// On any endpoint when auth is required (eg. home route)
+...
+import { authentication } from "@co-digital/login";
+
+const homePageRouter = Router();
+homePageRouter.get(config.HOME_URL, authentication, get);
+...
+```
+
+- **Frontend Integration**: `COLA` has already developed user interface components for login, logout, and other authentication-related actions. However, for integration with other Identity Providers (IdPs) such as `OneLogin`, you'll need to implement these components within your application.
+
+## Note
+
+- `COLA` provides a secure login mechanism for any Cabinet Office service that hosts under the some domain - `cabinetoffice.gov.uk` and under the hood utilises and abstracts AWS Cognito for user management.
```

Comment:
Unclear wording/structure, something like:

"... service that is hosted under the `cabinetoffice.gov.uk` domain. Under the hood, it utilises and abstracts AWS Cognito for user management."